### DISCUSSION TITLE
Conditional route based global css in _app.js

### DISCUSSION THREAD

**Discussion by vikasg603** [vercel/next.js#25925] [Ideas]
### Describe the feature you'd like to request

I have searched for many articles and previous issues regarding conditional global CSS, but I cannot get any.
There is 1 solution for this using custom CSS loader using styled-jsx, but again, the problem with styled-jsx is it doesn't generate the separate external CSS file but rather creates internal CSS on the build page.

This is not good for caching when we have a large number of pages.

In short, I am looking for any way through which either we can import global CSS on component level or route-based conditional global CSS import.

### Describe the solution you'd like

Conditional global CSS importing, using useRouter() on _app.js

### Describe alternatives you've considered

Component-based global CSS importing, where we can import global CSS anywhere in the code.
